code/modules/mining/drill.dm

```diff
@@ -0,0 +1,379 @@
+		active = FALSE
+		soundloop.stop()
+		update_overlays()
+		update_icon_state()
+		return
+
+/obj/machinery/drill/Destroy()
+	QDEL_NULL(soundloop)
+	QDEL_NULL(cell)
+	return ..()
+
+//Instead of being qdeled the drill requires mildly expensive repairs to use again
+/obj/machinery/drill/deconstruct(disassembled)
+	if(active && mining)
+		say("Drill integrity failure. Engaging emergency shutdown procedure.")
+		//Just to make sure mobs don't spawn infinitely from the vein and as a failure state for players
+		mining.deconstruct()
+	obj_break()
+	update_icon_state()
+	update_overlays()
+
+/obj/machinery/drill/get_cell()
+	return cell
+
+//The RPED sort of trivializes a good deal of the malfunction mechancis, as such it will not be allowed to work
+/obj/machinery/drill/exchange_parts(mob/user, obj/item/storage/part_replacer/W)
+	to_chat(user, "<span class='notice'>[W] does not seem to work on [src], it might require more delicate part manipulation.")
+	return
+
+/obj/machinery/drill/attackby(obj/item/tool, mob/living/user, params)
+	var/obj/structure/vein/vein = locate(/obj/structure/vein) in src.loc
+	if(machine_stat & BROKEN)
+		if(istype(tool,/obj/item/stack/sheet/plasteel))
+			var/obj/item/stack/sheet/plasteel/plating = tool
+			if(plating.use(10,FALSE,TRUE))
+				metal_attached = METAL_PLACED
+				to_chat(user, "<span class='notice'>You prepare to attach the plating to [src].</span>")
+				return
+			else
+				to_chat(user, "<span class='notice'>You don't have enough plasteel to fix the plating.</span>")
+				return
+		if(metal_attached == METAL_SECURED && tool.tool_behaviour == TOOL_WELDER)
+			if(tool.use_tool(src, user, 30, volume=50))
+				to_chat(user, "<span class='notice'>You weld the new plating onto the [src], successfully repairing it.")
+				metal_attached = null
+				obj_integrity = max_integrity
+				set_machine_stat(machine_stat & ~BROKEN)
+				update_icon_state()
+				return
+	if(tool.tool_behaviour == TOOL_WRENCH)
+		if(metal_attached && machine_stat & BROKEN)
+			if(tool.use_tool(src, user, 30, volume=50))
+				to_chat(user, "<span class='notice'>You bolt the plating the plating in place on [src].</span>")
+				metal_attached = METAL_SECURED
+				return
+		if(!vein && !anchored)
+			to_chat(user, "<span class='notice'>[src] must be on top of an ore vein.</span>")
+			return
+		if(active)
+			to_chat(user, "<span class='notice'>[src] can't be unsecured while it's running!</span>")
+			return
+		if(!anchored && tool.use_tool(src, user, 30, volume=50))
+			to_chat(user, "<span class='notice'>You secure the [src] to the ore vein.</span>")
+			playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+			mining = vein
+			anchored = TRUE
+			update_icon_state()
+			return
+		if(tool.use_tool(src, user, 30, volume=50))
+			to_chat(user, "<span class='notice'>You unsecure the [src] from the ore vein.</span>")
+			playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+			anchored = FALSE
+			if(mining.spawner)
```

Comment:
```suggestion
			if(mining?.spawner)
```
This is a solution to a runtime on live, but it may be a bigger issue of mining being null when it shouldn't be. The runtime in question:

```
Cannot read null.spawner : 5
 | attackby (/obj/machinery/drill/attackby) : 5
 | - file: code/modules/mining/drill.dm,148
 | - usr: oc: the iron sand (147,238,5) (/turf/open/floor/plating/asteroid/rockplanet/lit)
 | - src: the core sampling research dri... (/obj/machinery/drill/mission)
 | - loc: the snow (148,238,5) (/turf/open/floor/plating/asteroid/snow/lit/rockplanet)
 | - stack: 
 |   - the core sampling research dri... (/obj/machinery/drill/mission): attackby(the wrench (/obj/item/wrench), Beedost Oktba (/mob/living/carbon/human), "icon-x=17;icon-y=14;left=1;but...")
 |   - the wrench (/obj/item/wrench): melee attack chain(Beedost Oktba (/mob/living/carbon/human), the core sampling research dri... (/obj/machinery/drill/mission), "icon-x=17;icon-y=14;left=1;but...")
 |   - Beedost Oktba (/mob/living/carbon/human): ClickOn(the core sampling research dri... (/obj/machinery/drill/mission), "icon-x=17;icon-y=14;left=1;but...")
 |   - the core sampling research dri... (/obj/machinery/drill/mission): Click(the snow (148,238,5) (/turf/open/floor/plating/asteroid/snow/lit/rockplanet), "mapwindow.map", "icon-x=17;icon-y=14;left=1;but...")

Reply:
Ough, I'm not sure how I never found this one while I was testing it initially, it's almost assuredly just null because the vein's been deleted so I don't think it's a more widespread issue, but it's such an insanely common occurrence I'm not sure how I never stumbled into it at least once.